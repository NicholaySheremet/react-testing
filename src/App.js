import {useEffect, useState} from 'react';

import './App.css';

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);

  const onClick = () => setToggle(val => !val);

  useEffect(() => {
    setTimeout(() => {
      setData({})
    }, 100);
  }, [])

  return (
    <div className="App">
      {
        toggle === true
        && <div data-testid='toggle-element'>toggled data</div>
      }
      {data && <div>test data</div>}
      <h1>React App</h1>
      <button data-testid='toggle-btn' onClick={onClick}>clicker</button>
      <input type='text' placeholder='input value'/>
    </div>
  );
}

export default App;
