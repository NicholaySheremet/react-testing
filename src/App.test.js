import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

describe('TEST APP', () => {

  test('renders react app components', () => {
    render(<App />);
    const reactElement = screen.getByText(/React App/i);
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/input value/i);
    expect(reactElement).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('renders react components', () => {
    render(<App />);
    const helloElement = screen.queryByText(/hello/i);
    expect(helloElement).toBeNull();
  });

  test('renders async react components', async () => {
    render(<App />);
    const element = await screen.findByText(/test data/i);
    expect(element).toBeInTheDocument();
  });

  test('button working', () => {
    render(<App />);
    const btn = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-element')).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-element')).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-element')).toBeNull();
  });
})
