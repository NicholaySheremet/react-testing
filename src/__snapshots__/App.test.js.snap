// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP renders react app components 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
